disk.go

```diff
@@ -23,6 +23,48 @@ func (t DiskType) String() string {
 	return []string{"HDD", "SSD", "NVME"}[t]
 }
 
+// StringToDiskType - convert a string to a disk type.
+func StringToDiskType(typeStr string) DiskType {
+	kmap := map[string]DiskType{
+		"HDD":  HDD,
+		"SSD":  SSD,
+		"NVME": NVME,
+	}
+	if dtype, ok := kmap[typeStr]; ok {
+		return dtype
+	}
+
+	return HDD
+}
+
+// MarshalJSON - Custom to marshal as a string.
+func (t *DiskType) MarshalJSON() ([]byte, error) {
```

Comment:
I would actually do MarshalText instead of MarshalJSON so that json/xml and any other text encoder can use it.

Reply:
I'll change. thank you.